transcendence-app/src/user/user.controller.spec.ts

```diff
@@ -6,32 +6,53 @@ import { UserModule } from './user.module';
 import { UserService } from './user.service';
 import { UserDto } from './dto/user.dto';
 import { NotFoundException } from '@nestjs/common';
-import { UserEntity } from './user.entity';
-import { Constants } from '../shared/enums/commonconsts.enum';
+import { AuthenticatedGuard } from '../shared/guards/authenticated.guard';
+import { v4 as uuidv4 } from 'uuid';
+
+const testUserDto: UserDto = {
+  provider: 'ft_transcendence',
+  image_url: 'www.example.jpg',
+  username: 'user',
+  email: 'afgv@github.com',
+};
+const testUserId = uuidv4();
 
 describe('User Controller end to end test', () => {
   let app: INestApplication;
+  const canActivate = jest.fn(() => true);
+
   beforeAll(async () => {
     const module: TestingModule = await Test.createTestingModule({
       imports: [UserModule],
       controllers: [UserController],
-    }).compile();
+    })
+      .overrideGuard(AuthenticatedGuard)
+      .useValue({ canActivate })
+      .compile();
 
     app = module.createNestApplication();
     await app.init();
   });
 
   it('should create a user', async () => {
     const server = await app.getHttpServer();
-    await request(server)
-      .post(
-        '/user/new?provider=42&username=afgv&email=hello@example.com&image_url=www.example.jpg',
-      )
+    const response = await request(server)
+      .post('/user')
+      .send(testUserDto)
```

Comment:
Is there a way to send this manually for testing purposes on some browser (e.g. firefox)? How do you manually set the body of the request to the Dto that you want to mock? How do you serialize that object? Why is it better than the query?

Reply:
For me, Postman is usually the way to go